### PR TITLE
Migrate figwheel conf from project.clj to figwheel ns

### LINKED ISSUE #2999: Simplify our figwheel config
### User Story

As a developer, I want to rely on a clean figwheel configuration

### Description

Our current figwheel conf is retrieved by some code from project.clj. This is pretty complex, brittle and makes `project.clj` unnecessarily big. Migrate relevant conf to the figwheel ns.

Tasks:
* remove lein-figwheel plugin
* migrate all figwheel conf from `project.clj` to `figwheel` ns

*Type*: Feature

### PR DESCRIPTION
**Pull request by voxlet**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

Fixes #2999

### Summary:

[comment]: # (Summarise the problem and how the pull request solves it)
Replaced the figwheel profile :cljsbuild section with the equivalent in figwheel.edn format (along with the config reshaping code in figwheel-api), making sure not to change the behavior of `figwheel-api/start`.

### Review notes (optional):

I erred on the conservative side with the changes, preserving the *-test builds and warning-handler in the android build.

### Steps to test:
- Confirm `lein figwheel-repl` behavior has not changed

status: ready

### CODE REVIEW

File: env/dev/figwheel.clj [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+(defn get-id [id]
+  (-> id
+      name
+      (s/replace #"-test" "")
+      keyword))
+
+(defn get-build [builds id]
+  (when-let [build (first (filter #(= (:id %) (get-id id)) builds))]
+    (if (test-id? id)
```
Comment (jeluard) [inline env/dev/figwheel.clj]: Is this part still relevant? If not we could simply remove it.

**Review by jeluard — COMMENTED** [review]
Thanks looks great!

**Review by dmitryn — APPROVED** [review]
Thanks for the contribution!

File: env/dev/figwheel.clj (hunk 1 above)
Comment (voxlet) [reply env/dev/figwheel.clj]: Not that I can see, at least in terms of immediate code dependency. It was kept for the case of everyday REPL session use, but as far as I can tell the *-test runs are already broken in develop. 

Maybe we can go ahead and delete the code, and see if there any complaints? I think the chances of that happening are low enough.

Comment (jeluard) [reply env/dev/figwheel.clj]: Please remove it then! :)

**Comment by voxlet** [comment]
@jeluard Thanks for the feedback, removed the test builds. 
Also migrated the key names to the external figwheel.edn format (described [here](https://github.com/bhauman/lein-figwheel/blob/0f62d6d043abb6156393fd167f6c1496c5439689/sidecar/src/figwheel_sidecar/schemas/config.clj#L47-L74)), and moved the static nREPL port config there as well.

**Review by jeluard — APPROVED** [review]

**Comment by flexsurfer** [comment]
@voxlet please squash commits

**Pull request merged**